test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -262,6 +262,28 @@ Describe "Import-Module for Binary Modules" -Tags 'CI' {
     }
 }
 
+Describe "PSModuleAutoloadingPreference has correct default value and rejects bogus values" -Tags 'CI' {
+
+    It "Default value should be all" {
+        $Global:PSModuleAutoloadingPreference | Should -Be "All"
+    }
+
+    It "Assign bogus value fails" {
+        {$Global:PSModuleAutoloadingPreference = "bogus" } | Should -Throw -ErrorId "RuntimeException"
+    }
```

Comment:
Should we add tests for empty string and numeric?
Should/can we have more useful ErrorId?